Jenkinsfile

```diff
@@ -0,0 +1,128 @@
+label = "${UUID.randomUUID().toString()}"
+BUILD_FOLDER = "/go"
+docker_user = "iguaziodocker"
+docker_credentials = "iguazio-prod-docker-credentials"
+git_project = "iguazio_api_examples"
+git_project_user = "v3io"
+git_deploy_user = "iguazio-prod-git-user"
+git_deploy_user_token = "iguazio-prod-git-user-token"
+
+properties([pipelineTriggers([[$class: 'PeriodicFolderTrigger', interval: '2m']])])
+podTemplate(label: "${git_project}-${label}", yaml: """
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "${git_project}-${label}"
+  labels:
+    jenkins/kube-default: "true"
+    app: "jenkins"
+    component: "agent"
+spec:
+  shareProcessNamespace: true
+  containers:
+    - name: jnlp
+      image: jenkinsci/jnlp-slave
+      resources:
+        limits:
+          cpu: 1
+          memory: 2Gi
+        requests:
+          cpu: 1
+          memory: 2Gi
+      volumeMounts:
+        - name: go-shared
+          mountPath: /go
+    - name: docker-cmd
+      image: docker
+      command: [ "/bin/sh", "-c", "--" ]
+      args: [ "while true; do sleep 30; done;" ]
+      volumeMounts:
+        - name: docker-sock
+          mountPath: /var/run
+        - name: go-shared
+          mountPath: /go
+  volumes:
+    - name: docker-sock
+      hostPath:
+          path: /var/run
+    - name: go-shared
+      emptyDir: {}
+"""
+    ) {
+    node("${git_project}-${label}") {
+        withCredentials([
+                usernamePassword(credentialsId: git_deploy_user, passwordVariable: 'GIT_PASSWORD', usernameVariable: 'GIT_USERNAME'),
+                string(credentialsId: git_deploy_user_token, variable: 'GIT_TOKEN')
+        ]) {
+            def AUTO_TAG
+            def TAG_VERSION
+
+            stage('get tag data') {
+                container('jnlp') {
+                    TAG_VERSION = sh(
+                            script: "echo ${TAG_NAME} | tr -d '\\n' | egrep '^v[\\.0-9]*.*\$' | sed 's/v//'",
+                            returnStdout: true
+                    ).trim()
+
+                    sh "curl -v -H \"Authorization: token ${GIT_TOKEN}\" https://api.github.com/repos/${git_project_user}/${git_project}/releases/tags/v${TAG_VERSION} > ~/tag_version"
```

Comment:
common.shell instead of sh
but also check if maybe there is a plugin that do some github integration and help with this
since i am not sure from where TAG_NAME is coming, then need to make sure that tag_VERSION is indeed a result of TAG_NAME & not to identical by accident or pure luck